PocketKit/Sources/PocketKit/Navigation/NavigationSidebarViewController.swift

```diff
@@ -0,0 +1,98 @@
+        MainViewModel.AppSection.allCases.forEach {
+            snapshot.appendItems([$0.navigationTitle], toSection: $0)
+        }
+        return snapshot
+    }()
+    
+    private lazy var dataSource: UICollectionViewDiffableDataSource<MainViewModel.AppSection, String> = {
+        let registration = UICollectionView.CellRegistration<UICollectionViewListCell, String> { (cell, indexPath, string) in
+            var content = cell.defaultContentConfiguration()
+            content.attributedText = NSAttributedString(string, style: .title)
+            cell.contentConfiguration = content
+        }
+        
+        return UICollectionViewDiffableDataSource<MainViewModel.AppSection, String>(
+            collectionView: collectionView
+        ) { (collectionView, indexPath, string) -> UICollectionViewCell? in
+            let cell = collectionView.dequeueConfiguredReusableCell(
+                using: registration,
+                for: indexPath,
+                item: string
+            )
+            
+            let primaryColor = UIColor(.ui.grey6)
+            
+            let backgroundView = UIView()
+            backgroundView.backgroundColor = primaryColor.withAlphaComponent(0.2)
+            cell.backgroundView = backgroundView
+            
+            let selectedBackgroundView = UIView()
+            selectedBackgroundView.backgroundColor = primaryColor
+            cell.selectedBackgroundView = selectedBackgroundView
+            
+            return cell
+        }
+    }()
+    
+    private lazy var collectionView: UICollectionView = UICollectionView(
+        frame: .zero,
+        collectionViewLayout: Self.layout
+    )
+    private let model: MainViewModel
+    private var subscriptions: Set<AnyCancellable> = []
+
+    init(model: MainViewModel) {
+        self.model = model
+        super.init(nibName: nil, bundle: nil)
+        
+        self.title = "Pocket"
+        
+        collectionView.alwaysBounceVertical = true
+        collectionView.delegate = self
+        collectionView.dataSource = dataSource
+        dataSource.apply(Self.snapshot)
+        
+        model.$selectedSection
+            .receive(on: DispatchQueue.main)
+            .sink { [weak self] section in
+                let indexPath = self?.dataSource.indexPath(for: section.navigationTitle)
+                self?.collectionView.selectItem(at: indexPath, animated: false, scrollPosition: .top)
+            }
+            .store(in: &subscriptions)
+    }
+    
+    override func loadView() {
+        view = collectionView
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) is not implemented")
+    }
+}
+
+extension NavigationSidebarViewController: UICollectionViewDelegate {
+    func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
+        let section = dataSource.snapshot().sectionIdentifiers[indexPath.section]
```

Comment:
See above note regarding building the snapshot, but we should be able to set the app section like so:

```swift
model.selectedSection = MainViewModel.AppSection.allCases[indexPath.item]
```